cordova-lib/src/cordova/platform.js

```diff
@@ -110,15 +110,25 @@ function addHelper(cmd, hooksRunner, projectRoot, targets, opts) {
                 if(platform === 'wp8') {
                     events.emit('warn', 'wp8 has been deprecated. Please use windows instead.');
                 }
-                if(spec && pkgJson) {
-                    pkgJson.dependencies[pkgJsonSpec] = spec;
+                // If there is a spec specified during add, add to pkg.json dependencies(if exists).
```

Comment:
Instead, I found that this worked better. Instead of using semver satisfies, I just check to see if the spec and what is in pkg.json actually match up and just write if they do not. My tests are able to pass this way. What do you think? Am I way off?

```                 if(spec && pkgJson && pkgJson.dependencies && (pkgJson.dependencies[prefixCordovaPlatform] || pkgJson.dependencies[platform])) {
                    if (spec === pkgJson.dependencies[prefixCordovaPlatform] && spec === pkgJson.dependencies[platform]) {
                    } else {
                        if (pkgJson.dependencies[prefixCordovaPlatform]) {
                            pkgJson.dependencies[prefixCordovaPlatform] = '^'+spec;
                        } else if (pkgJson.dependencies[platform]) {
                            pkgJson.dependencies[platform] = '^'+spec;
                        }
                        fs.writeFileSync(pkgJsonPath, JSON.stringify(pkgJson, null, 4), 'utf8');
                    }
                } ```